newIDE/app/src/AnnouncementsFeed/AnnouncementsFeedContext.js

```diff
@@ -31,35 +41,43 @@ export const AnnouncementsFeedStateProvider = ({
     null
   );
   const [error, setError] = React.useState<?Error>(null);
+  const [promotions, setPromotions] = React.useState<?(Promotion[])>(null);
   const isLoading = React.useRef<boolean>(false);
 
-  const fetchAnnouncements = React.useCallback(
-    () => {
-      if (isLoading.current) return;
+  const fetchAnnouncementsAndPromotions = React.useCallback(() => {
+    if (isLoading.current) return;
 
-      (async () => {
-        setError(null);
-        isLoading.current = true;
+    (async () => {
+      setError(null);
+      isLoading.current = true;
 
-        try {
-          const announcements = await listAllAnnouncements();
+      try {
+        const [fetchedAnnouncements, fetchedPromotions] = await Promise.all([
+          listAllAnnouncements(),
+          listAllPromotions(),
+        ]);
 
-          setAnnouncements(announcements);
-        } catch (error) {
-          console.error(
-            `Unable to load the announcements from the api:`,
-            error
-          );
-          setError(error);
-        }
+        // Logic to remove once promotions are displayed to enough users.
+        // For now, we filter out promotions from the announcements.
+        const filteredAnnouncements = fetchedAnnouncements.filter(
```

Comment:
Should you add the check we do a bit everywhere "Array.isArray"?

Reply:
Probably should be in the function doing the actual API call